src/openfhe.jl

```diff
@@ -117,576 +209,568 @@ function init_bootstrapping!(context::SecureContext{<:OpenFHEBackend},
+function add(sa1::SecureArray{<:OpenFHEBackend}, sa2::SecureArray{<:OpenFHEBackend})
+    cc = get_crypto_context(sa1)
+    ciphertexts = Vector{OpenFHE.Ciphertext}(undef, length(sa1.data))
+    for i in range(1, length(sa1.data))
+        ciphertexts[i] = OpenFHE.EvalAdd(cc, sa1.data[i], sa2.data[i])
+    end
+    secure_array = SecureArray(ciphertexts, size(sa1), capacity(sa1), sa1.context)
 
-    secure_vector
+    secure_array
 end
 
-function add(sv::SecureVector{<:OpenFHEBackend}, pv::PlainVector{<:OpenFHEBackend})
-    cc = get_crypto_context(sv)
-    ciphertext = OpenFHE.EvalAdd(cc, sv.data, pv.data)
-    secure_vector = SecureVector(ciphertext, length(sv), capacity(sv), sv.context)
+function add(sa::SecureArray{<:OpenFHEBackend}, pa::PlainArray{<:OpenFHEBackend})
+    cc = get_crypto_context(sa)
+    ciphertexts = Vector{OpenFHE.Ciphertext}(undef, length(sa.data))
+    for i in range(1, length(sa.data))
+        ciphertexts[i] = OpenFHE.EvalAdd(cc, sa.data[i], pa.data[i])
+    end
+    secure_array = SecureArray(ciphertexts, size(sa), capacity(sa), sa.context)
 
-    secure_vector
+    secure_array
 end
 
-function add(sv::SecureVector{<:OpenFHEBackend}, scalar::Real)
-    cc = get_crypto_context(sv)
-    ciphertext = OpenFHE.EvalAdd(cc, sv.data, scalar)
-    secure_vector = SecureVector(ciphertext, length(sv), capacity(sv), sv.context)
+function add(sa::SecureArray{<:OpenFHEBackend}, scalar::Real)
+    cc = get_crypto_context(sa)
+    ciphertexts = Vector{OpenFHE.Ciphertext}(undef, length(sa.data))
+    for i in range(1, length(sa.data))
+        ciphertexts[i] = OpenFHE.EvalAdd(cc, sa.data[i], scalar)
+    end
+    secure_array = SecureArray(ciphertexts, size(sa), capacity(sa), sa.context)
 
-    secure_vector
+    secure_array
 end
 
-function subtract(sv1::SecureVector{<:OpenFHEBackend}, sv2::SecureVector{<:OpenFHEBackend})
-    cc = get_crypto_context(sv1)
-    ciphertext = OpenFHE.EvalSub(cc, sv1.data, sv2.data)
-    secure_vector = SecureVector(ciphertext, length(sv1), capacity(sv1), sv1.context)
+function subtract(sa1::SecureArray{<:OpenFHEBackend}, sa2::SecureArray{<:OpenFHEBackend})
+    cc = get_crypto_context(sa1)
+    ciphertexts = Vector{OpenFHE.Ciphertext}(undef, length(sa1.data))
+    for i in range(1, length(sa1.data))
+        ciphertexts[i] = OpenFHE.EvalSub(cc, sa1.data[i], sa2.data[i])
+    end
+    secure_array = SecureArray(ciphertexts, size(sa1), capacity(sa1), sa1.context)
 
-    secure_vector
+    secure_array
 end
 
-function subtract(sv::SecureVector{<:OpenFHEBackend}, pv::PlainVector{<:OpenFHEBackend})
-    cc = get_crypto_context(sv)
-    ciphertext = OpenFHE.EvalSub(cc, sv.data, pv.data)
-    secure_vector = SecureVector(ciphertext, length(sv), capacity(sv), sv.context)
+function subtract(sa::SecureArray{<:OpenFHEBackend}, pa::PlainArray{<:OpenFHEBackend})
+    cc = get_crypto_context(sa)
+    ciphertexts = Vector{OpenFHE.Ciphertext}(undef, length(sa.data))
+    for i in range(1, length(sa.data))
```

Comment:
This is a highly unusual style for loop iterations. Can you please use
```suggestion
    for i in 1:length(sa.data)
```
here and everywhere else? Even more Julian would be to use
```suggestion
    for i in eachindex(sa.data)
```

But please do not iterate over ranges, this is not Python 😅